### DISCUSSION TITLE
Broken VM Virtual Drive - fixed!

### DISCUSSION THREAD

**Discussion by LucasTheIntern** [bitcraze/discussions#1394] [Q&A]
Hi!

As the title implies, I have somehow broken a copy of the VM. It sounds like there are a lot of things that might have caused this, but I think it was caused by my laptop powering off while I was away because I left it unplugged. :( 

I've had this happen multiple times and always just made a new VM when it refused to boot. But this time I had some files that I REALLY wanted to get back, so I figured out a fix. 

Trying to start the VM brings up a window in the Oracle Virtual Box Manager. I see the Virtual Box logo for a few seconds, and then the terminal that usually shows messages while the Ubuntu system is booting. It starts doing its thing, and then I get the following:
```

Begin: Running /scrips/init-premount ... done.
Begin: Mounting root file system ... Begin: Running /scripts/local-top ... done.
Begin: Running /scripts/local-premount ... [    3.600029] Btrfs loaded, crc32c=crc32c-intel, zoned=yes, fsverity=yes
Scanning for Btrfs filesystems
done.
Begin: Will now check root file system ... fsck from util-linux 2.37.2
[/usr/sbin/fsck.ext4 (1) -- /dev/mapper/ubuntu--vg-ubuntu--1v] fsck.ext4 -a -C0 /dev/mapper/ubuntu--vg-ubuntu==1v
/dev/mapper/ubuntu--vg-ubuntu--1v contains a file system with errors, check forced. 
/dev/mapper/ubuntu--vg-ubuntu--1v:
Inodes that were part of a corrupted orphan linked list found.

/dev/mapper/ubuntu--vg-ubuntu--1v: UNEXPECTED INCONSISTENCY; RUN fsck MANUALLY.
    (i.e., without -a or -p options)
fsck exited with status code 4
done. 
Failure: File system check of the root filesystem failed
The root filesystem on /dev/mapper/ubuntu--vg-ubuntu--1v requires a manual fsck

BusyBox v1.30.1 (Ubuntu 1:1.30.1-7ubuntu3) built-in shell (ash)
Enter 'help' for a list of built-in commands.

(Initramfs) _

```
Seems like related to the issue of not removing usb drives before "ejecting" them, but that's all I've got. 

I was hoping that there is some way to either:
1. Fix the VM and get it running again
2. Rescue items from the .vdi file by mounting it to a new VM

**Method 1:**
I followed the answer by user "spamegg" in this forum for a similar issue:
https://forums.linuxmint.com/viewtopic.php?t=354549#:~:text=You%20can%20run%20it%20with%20fsck%20%2Fdev%2Fsda8.%20After,will%20not%20be%20mounted%20yet%2C%20so%20it%27s%20OK.

I ended up running the following command:
`exec fsck /dev/mapper/ubuntu--vg-ubuntu--lv`
and then just answering yes to all of the suggested fixes. The VM booted up normally and I am able to open my files!

**Method 2:**
I still haven't tried this - it seems a bit too technical for my lack of IT expertise. I found plenty of forums that mentioned mounting .vdi's, but I didn't find any instructions on _how_. If anyone knows how to do this (or if it's even a good idea), feel free to leave an answer! I have a hunch that the "corrupted orphan linked list" issue would persist though.

**Comment by ataffanel** [comment]
Hi!

Thanks a lot for sharing a solution! This is off that this problem happens so often, I would not expect the Linux filesystem to corrupts so easily. Maybe the VM is not helping ... Anyway this is definitely information that can be useful to someone else in the same situation so thanks for sharing!
